src/cc_catalog_airflow/dags/util/loader/sql.py

```diff
@@ -390,3 +391,59 @@ def update_flickr_sub_providers(
     Drop the temporary table
     """
     postgres.run(f'DROP TABLE public.{temp_table};')
+
+
+def update_europeana_sub_providers(
+  postgres_conn_id,
+  image_table=IMAGE_TABLE_NAME,
+  default_provider=prov.EUROPEANA_DEFAULT_PROVIDER,
+  sub_providers=prov.EUROPEANA_SUB_PROVIDERS
+):
+    postgres = PostgresHook(postgres_conn_id=postgres_conn_id)
+
+    select_query = dedent(
+        f'''
+        SELECT
+        {col.FOREIGN_ID} AS foreign_id,
+        {col.META_DATA} ->> 'dataProvider' AS data_providers
+        FROM {image_table}
+        WHERE {col.PROVIDER} = '{default_provider}';
+        '''
+    )
+
+    selected_records = postgres.get_records(select_query)
+
+    """
+    Filter the records to retain the ones with desired sub providers.
+    Each record in the new list would contain the sub provider name at the end.
+    """
+    filtered_records = []
+    for row in selected_records:
+        data_providers = ast.literal_eval(row[1])
```

Comment:
I think it would be better to use `json.loads` for this.  `ast.literal_eval` will throw an Exception if a null appears in any result array, whereas `json.loads` will correctly translate it into a `Nonetype`